pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -509,6 +522,30 @@ func (wm Watermark) GetMaxDelay() time.Duration {
 	return time.Duration(0)
 }
 
+func (wm Watermark) GetIdleSource() *IdleSource {
+	if wm.IdleSource != nil {
+		return wm.IdleSource
+	}
+
+	return nil
+}
+
+func (is IdleSource) GetMaxWait() time.Duration {
+	if is.MaxWait != nil {
+		return is.MaxWait.Duration
+	}
+
+	return time.Duration(-1)
```

Comment:
The default value is different from `-1s` defined above.

Reply:
I'll update it, `-1` as a thought to put initially as a default value but I don't see any benefits for it.